### DISCUSSION TITLE
[BUG] On customer completing checkout cart does not get deleted

### DISCUSSION THREAD

**Discussion by magnetism2000** [Shopify/storefront-api-feedback#20] [Help] [answered]
**Problem:**
Cart is not being deleted when the cart has been paid for and completed.

According to these docs, the cart should be deleted when a user checkouts.
https://shopify.dev/custom-storefronts/cart#considerations
_'Shopify automatically deletes the cart when the customer completes their checkout.'_

When the user navigates back to the domain of the headless site, the site fetches the cart using the CartId then that completed cart is what is shown in their basket. 

Is there any way to check if the cart is completed using the cart API?

**Comment by maddalax** [comment]
I am also having this same issue.

**Reply by morganmccunn** [reply]
Hey folks! We appreciate you documenting this! We're aware of this limitation and working on it. I'll report back!

**Comment by cocoahero** [comment, marked as answer]
Hi @oscarmanderj,

A fix for this has been deployed. Can you please test again and see if the issue persists? Note that this fix will only effect new carts / checkouts.

Thanks again for the report!

**Reply by joepagan** [reply]
Hi @cocoahero @morganmccunn it appears that there is still a bug for carts which complete via the "Express Checkout" "Shop Pay" & "G Pay" options, Shopify GQL API will still return the cart in that scenario.

I have not tested all the other Express options as they are not enabled on this store but I suspect given 2 of 2 are not working correctly, it is worth checking the others.

To confirm, the standard (non-express) checkout option does return null after checkout completes.

This is causing issues where repeat customers are processing incorrect orders as they will add new products to an old cart, so I would appreciate some support.

**Update 2023-01-18:
[Resolved](https://github.com/Shopify/storefront-api-feedback/discussions/12#discussioncomment-4720257)**
